tests/stub/driver_parameters/test_bookmark_manager.py

```diff
@@ -474,16 +474,182 @@ def test_should_resolve_database_name_with_system_bookmarks(self):
     def _new_driver(self, bookmark_manager_config=None):
         if bookmark_manager_config is None:
             bookmark_manager_config = Neo4jBookmarkManagerConfig()
-
         uri = "neo4j://%s" % self._router.address
         auth = types.AuthorizationToken("basic", principal="neo4j",
                                         credentials="pass")
         return Driver(
             self._backend,
             uri, auth,
-            bookmark_manager_config=bookmark_manager_config
+            bookmark_manager_config=bookmark_manager_config)
+
+    def test_should_call_bookmark_supplier_for_all_get_bookmarks_call(self):
+        self._start_server(self._router, "router_with_db_name.script")
+        self._start_server(self._server, "transaction_chaining.script")
+
+        adb_bookmarks = ["adb:bm1"]
+        get_bookmarks_calls = []
+
+        def get_bookmarks(db):
+            get_bookmarks_calls.append([db])
+            return []
+
+        self._driver = self._new_driver(
+            Neo4jBookmarkManagerConfig(
+                initial_bookmarks={
+                    "adb": adb_bookmarks
+                },
+                bookmarks_supplier=get_bookmarks
+            )
+        )
+
+        s1 = self._driver.session("w", database="neo4j")
+        tx1 = s1.begin_transaction({"return_bookmark": "bm1"})
```

Comment:
```suggestion
        tx1 = s1.begin_transaction(tx_meta={"return_bookmark": "bm1"})
```

And in all other places. I actually had to look up what the dict was doing.